socket_windows.go

```diff
@@ -0,0 +1,181 @@
+package kr
```

Comment:
I've consolidated common code in socket.c with OS-specific AgentListen, DaemonListen, HostAuthListen, and DaemonDial.  As discussed via phone there is still some significant cut-n-paste between socket_linux and socket_darwin.   I suggest we merge this pull request then go back and tidy that up.

The majority of commit 4 is actually related to getting all of the unit tests to run on windows.  It was notably intrusive to fix pairOver() and unPairOver() because they were coded to accept a unix socket name.   The fix required significant refactoring to always use the sockets abstraction.

I've tested on windows and on linux.  I haven't tested on darwin (because I don't have a mac).